### PR TITLE
[Bugfix] DeepSeek Accuracy

### PR DESCRIPTION
**Pull request by LucasWilkinson**
Fix accuracy issue introduced by https://github.com/vllm-project/vllm/pull/14384 , on main `forward_native` is called (due to `enabled()` be false in `CustomOp(nn.Module)` on main) the PR updated this cuda, changing the behavior slightly. We should probably investigate further to figure out why `forward_native` differs form `forward_cuda` (on an Nvidia GPUs). This is potentially due to `q_pe` and `k_pe` being different shapes in MLA. 

Main
```
local-completions (model=deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct,base_url=http://127.0.0.1:8192/v1/completions,num_concurrent=5,max_retries=3,tokenized_requests=False), gen_kwargs: (None), limit: 100.0, num_fewshot: None, batch_size: 1
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  | 0.06|±  |0.0239|
|     |       |strict-match    |     5|exact_match|↑  | 0.00|±  |0.0000|
```

This PR:
```
local-completions (model=deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct,base_url=http://127.0.0.1:8192/v1/completions,num_concurrent=5,max_retries=3,tokenized_requests=False), gen_kwargs: (None), limit: 100.0, num_fewshot: None, batch_size: 1
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  | 0.78|±  |0.0416|
|     |       |strict-match    |     5|exact_match|↑  | 0.77|±  |0.0423|
```

### CODE REVIEW

**Comment by LucasWilkinson** [comment]
Unfortunately gives up some of the gains:

```
  backend  input_tokens  output_tokens  output_toks/s     req/s  median_itl_ms  median_ttft_ms
3    vllm          1000           1000     982.249267  0.982249      42.002411     2403.797040
2    vllm          5000           1000     521.687045  0.521687      38.893414     6001.047062
4    vllm         10000           1000     331.643844  0.331644      36.228126    54635.778229
1    vllm         32000           1000     113.265287  0.113265      36.442538   183086.203407
```

Compared to: https://github.com/vllm-project/vllm/pull/14384#issuecomment-2705188465

**Review by tlrmchlsmth — APPROVED** [review]
very nice find

**Comment by LucasWilkinson** [comment]
Full revert was performed here: https://github.com/vllm-project/vllm/commit/ca7a2d5f28eac9621474563cdda0e08596222755 this is a partial revert

**Comment by xiaoqi35** [comment]
> Unfortunately gives up some of the gains:
> 
> ```
>   backend  input_tokens  output_tokens  output_toks/s     req/s  median_itl_ms  median_ttft_ms
> 3    vllm          1000           1000     982.249267  0.982249      42.002411     2403.797040
> 2    vllm          5000           1000     521.687045  0.521687      38.893414     6001.047062
> 4    vllm         10000           1000     331.643844  0.331644      36.228126    54635.778229
> 1    vllm         32000           1000     113.265287  0.113265      36.442538   183086.203407
> ```
> 
> Compared to: [#14384 (comment)](https://github.com/vllm-project/vllm/pull/14384#issuecomment-2705188465)

Could you show out the hardware information?

**Comment by LucasWilkinson** [comment]
8xH200, although this numbers are not that relevant after: https://github.com/vllm-project/vllm/pull/14540
